### PR TITLE
nit: Use sdk blocks instead of tendermint blocks.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Reference: https://github.com/cosmos/cosmos-sdk/issues/3544

basically saw it since its the only thing my editor is currently complaining about

### Commit Message / Changelog Entry

```bash
nit: Use sdk blocks instead of tendermint blocks.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/testsuite/grpc_query.go [hunk 1]
```diff
@@ -178,7 +177,7 @@ func (s *E2ETestSuite) QueryProposalV1(ctx context.Context, chain ibc.Chain, pro
 
 // GetBlockByHeight fetches the block at a given height. Note: we are explicitly using the res.Block type which has been
 // deprecated instead of res.SdkBlock to support backwards compatibility tests.
```
Comment (chatton) [inline e2e/testsuite/grpc_query.go:179]: I'm just noticing this comment.

**Review by chatton — CHANGES_REQUESTED** [review]
I just noticed the comment about why we chose to use `res.Block` instead of `res.SdkBlock`. I think we will need to double check the versions we support and when we can actually make this change.

File: e2e/testsuite/grpc_query.go (hunk 1 above)
Comment (DimitrisJim) [reply e2e/testsuite/grpc_query.go:179]: ahhh, I knew there was a gotcha, can't believe I missed it :face_exhaling:

**Review by chatton — COMMENTED** [review]
I won't leave an approval as I now have some commits here, but I think this new interface type should do the trick!

File: e2e/testsuite/grpc_query.go [hunk 2]
```diff
@@ -269,7 +275,13 @@ func (s *E2ETestSuite) GetBlockByHeight(ctx context.Context, chain ibc.Chain, he
...
 	}
 
-	return res.Block, nil
+	// versions newer than 0.47 SDK have the SdkBlock field.
+	if res.SdkBlock != nil {
+		return &res.SdkBlock.Header, nil
+	}
+
+	// older versions of the SDK do not have the SdkBlock field.
+	return &res.Block.Header, nil
```
Comment (colin-axner) [inline e2e/testsuite/grpc_query.go:284]: should we note somewhere that once support for v6 stops, this could be removed (maybe the interface as well?)?

**Review by colin-axner — APPROVED** [review]
LGTM

File: e2e/testsuite/grpc_query.go (hunk 2 above)
Comment (damiannolan) [reply e2e/testsuite/grpc_query.go:284]: Agree! Could we create an issue and add a link to it in a code comment somewhere around here?

**Review by damiannolan — APPROVED** [review]

File: e2e/testsuite/grpc_query.go (hunk 2 above)
Comment (DimitrisJim) [reply e2e/testsuite/grpc_query.go:284]: will do!

**Pull request merged**